authors/apps/articles/serializers.py

```diff
@@ -31,3 +32,37 @@ class Meta:
         model = Article
         fields = ('slug', 'title', 'description',
                   'body', 'image', 'created_at', 'updated_at')
+
+
+class RatingSerializer(serializers.ModelSerializer):
+    """Serializers for the rating model"""
+    max_rating = 5
+    min_rating = 1
+    user_rating = serializers.FloatField(
+        required=True,
+        validators=[
+            MinValueValidator(
+                min_rating, message="The minimum allowed rating is 1"),
+            MaxValueValidator(
+                max_rating, message="The maximum allowed rating is 5")
+        ],
+        error_messages={
+            "required": "Please provide a rating between 1 and 5"
+        }
+    )
+    article = serializers.SerializerMethodField()
+    average_rating = serializers.SerializerMethodField()
+
+    def get_average_rating(self, obj):
+        """Returns the average rating for an article"""
+        average_rating = Rating.objects.filter(
+            article=obj.article).aggregate(Avg('user_rating'))
+        return average_rating["user_rating__avg"]
+
+    def get_article(self, obj):
+        """Returns an article that matches the slug"""
+        return obj.article.slug
```

Comment:
Will this return the Article or the article's slug?

Reply:
It will return the article's slug. Let me change the variable name and docstring